challenge_spr/src/challenge_spr_states/riddle_game.py

```diff
@@ -171,22 +177,41 @@ def answer_placement_location(action):
 
 def answer_count_placement(action):
     entity = action['entity']
+    location = action['location']
+
     locations = [loc for loc in common_knowledge.locations if loc['name'] == entity]
-    if len(locations) == 1:
-        room = locations[0]['room']
-        if loc == action['location']:
-            return 'There is one %s in the %s' % (entity, room)
-        else:
-            return 'There are no %s in the %s' % (entity, room)
-    else:
+    if not locations:
         return 'I should count but I dont know that object %s' % entity
 
+    locations = [loc for loc in locations if loc['room'] == location]
+    if not locations:
+        return 'There are no %s in the %s' % (entity, location)
+    elif len(locations) == 1:
+        return 'There is one %s in the %s' % (entity, location)
+    else:
+        return 'There are %d %s in the %s' % (len(locations), entity, location)
+
+
+def answer_count_objects_placement(action):
+    entity = action['entity']
+    location = action['location']
+
+    objects = [obj for obj in common_knowledge.objects if obj['name'] == entity]
+    if not objects:
+        return "I should count but I don't know that object %s" % entity
+
+    if objects[0]["category"] in common_knowledge.category_locations:
+        (found_location, area_name) = common_knowledge.get_object_category_location(objects[0]["category"])
+        if found_location == location:
+            return 'There is one %s in the %s' % (entity, location)
+    return 'There are no %s in the %s' % (entity, location)
+
+
 def answer_find_objects(action):
     entity = action['entity']
     objects = [obj for obj in common_knowledge.objects if obj['name'] == entity]
     if len(objects) == 1:
         cat = objects[0]['category']
-        print cat
```

Comment:
Remove plain prints. If you need to report to the user, use logging tools. If you don't, then don't print at all.